File: LeetCode/1346-check-if-n-and-its-double-exist/1346-check-if-n-and-its-double-exist.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        mp = {}
        for x in arr:
            if x not in mp:
                mp[x] = 1
            else:
                mp[x] += 1
        for x in arr:
            if 2 * x in mp:
                if x != 0 or mp[0] > 1:
                    return True
        return False

File: LeetCode/1346-check-if-n-and-its-double-exist/1346-check-if-n-and-its-double-exist.cpp
class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        unordered_map<int, int> mp;
        for (int x : arr) mp[x]++;
        for (int x : arr) {
            if (mp.find(2 * x) != mp.end()) {
                if (x != 0 || mp[0] > 1) {
                    return true;
                }
            }
        }
        return false;
    }
};